pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/geospatial/MutableH3Index.java

```diff
@@ -18,48 +18,53 @@
  */
 package org.apache.pinot.core.realtime.impl.geospatial;
 
-import com.uber.h3core.H3Core;
+import com.google.common.base.Preconditions;
 import java.io.IOException;
-import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import org.apache.pinot.core.realtime.impl.ThreadSafeMutableRoaringBitmap;
-import org.apache.pinot.core.segment.creator.GeoSpatialIndexCreator;
-import org.apache.pinot.core.segment.creator.impl.geospatial.H3IndexResolution;
+import org.apache.pinot.core.segment.creator.impl.inv.geospatial.H3IndexResolution;
 import org.apache.pinot.core.segment.index.readers.H3IndexReader;
-import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+import org.apache.pinot.core.util.H3Utils;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
 import org.roaringbitmap.buffer.MutableRoaringBitmap;
 
 
 /**
  * A H3 index reader for the real-time H3 index values on the fly.
  * <p>This class is thread-safe for single writer multiple readers.
  */
-public class RealtimeH3IndexReader implements GeoSpatialIndexCreator, H3IndexReader {
-  private final H3Core _h3Core;
-  private final Map<Long, ThreadSafeMutableRoaringBitmap> _h3IndexMap = new ConcurrentHashMap<>();
+public class MutableH3Index implements H3IndexReader {
```

Comment:
shall this implement GeoSpatialIndexCreator?

Reply:
We don't mix mutable index with index creator. Index creator is for offline segment creation, and have a `seal()` method